src/c_ffi_support.rs

```diff
@@ -110,6 +121,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                         this.write_int(x, dest)?;
                         return Ok(());
                     }
+                    hir::PrimTy::Int(IntTy::Isize) => {
+                        let x = call::<isize>(ptr, &libffi_args.as_slice());
+                        if isize::BITS == 64 {
```

Comment:
I guess we should Add a check to the new -Z flag to make sure it's only used on the host and not for cross compiling, otherwise this line would need to check the target's bitwidth

Reply:
I thought we had a function for  writing an integer with a specific size... if not, create it and use it here and below

Reply:
Ah you're right -- [`write_int`](https://github.com/emarteca/miri/blob/int-function-args-returns/src/helpers.rs#L153) does take its integer argument and writes it to the size of `dest.layout.size`. The issue with `isize` and `usize` is that they don't satisfy the requirement of `impl Into<i128>` that `write_int` needs for its integer argument. 

I've just changed this to cast the `size`s to `i64` or `u64`, since this coverts both 32- and 64-bit machines.
I _think_ this means we also don't need to check the host vs target bitwidth, but correct me if I'm wrong!

Reply:
FWIW please don't use `as` casts in Miri, always go through `try_from`/`try_into` (or `from`/`into` if it is infallible).

We should really enable that clippy lint...